homeassistant/components/klyqa/datacoordinator.py

```diff
@@ -0,0 +1,224 @@
+                u_id = api.format_uid(group["id"])
+
+                light = [
+                    entity
+                    for entity in ha_entities
+                    if hasattr(entity, "u_id") and entity.u_id == u_id
+                ]
+
+                if len(light) == 0:
+                    # found klyqa device not in the light entities
+                    if (
+                        len(group["devices"]) > 0
+                        and "productId" in group["devices"][0]
+                        and group["devices"][0]["productId"].startswith(
+                            "@klyqa.lighting"
+                        )
+                    ):
+                        self.hass.bus.fire(EVENT_KLYQA_NEW_LIGHT_GROUP, group)
+        return True
+
+
+class KlyqaDataCoordinator(EntityComponent):
+    """KlyqaDataCoordinator."""
+
+    _instance = None
+    klyqa_accounts: dict[str, HAKlyqaAccount]
+    udp: socket.socket
+    tcp: socket.socket
+    remove_listeners: list
+    entries: dict[Any, Any]
+
+    # pylint: disable = super-init-not-called
+    def __init__(self) -> None:
+        """KlyqaDataCoordinator."""
+        raise RuntimeError("Call instance() instead")
+
+    def get_ports(self) -> int:
+        """Get_ports."""
+        try:
+            self.udp = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            self.udp.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
+            self.udp.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+            server_address = ("0.0.0.0", 2222)
+            self.udp.bind(server_address)
+            LOGGER.debug("Bound UDP port 2222")
+
+        except:  # noqa: E722 pylint: disable=bare-except
+            LOGGER.error(
+                "Error on opening and binding the udp port 2222 on host for initiating the lamp communication"
+            )
+            return 1
+
+        try:
+            self.tcp = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            self.tcp.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+            server_address = ("0.0.0.0", 3333)
+            self.tcp.bind(server_address)
+            LOGGER.debug("Bound TCP port 3333")
+            self.tcp.listen(1)
+
+        except:  # noqa: E722 pylint: disable=bare-except
+            LOGGER.error(
+                "Error on opening and binding the tcp port 3333 on host for initiating the lamp communication"
+            )
+            return 1
+        return 0
+
+    def init(
+        self,
+        logger: logging.Logger,
+        domain: str,
+        hass: HomeAssistant,
+        scan_interval: timedelta = DEFAULT_SCAN_INTERVAL,
+    ) -> None:
+        """__init."""
+        print("Init new instance")
+        super().__init__(logger, domain, hass, scan_interval)
+        self.klyqa_accounts = {}
+        self.get_ports()
+
+        self.entries = {}
+        self.remove_listeners = []
+
+    @classmethod
+    def instance(
```

Comment:
Please remove the use of a singleton. These are generally considered bad. Instead, create a single instance and store it for reuse.